aten/src/ATen/native/AffineGridGenerator.cpp

```diff
@@ -0,0 +1,97 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+
+namespace at { namespace native {
+
+at::Tensor linspace_from_neg_one(const Tensor& grid, int64_t val) {
+  if (val > 1) {
+    return at::linspace(-1, 1, val, grid.options());
+  } else {
+    return at::empty({}, grid.options()).fill_(-1);
```

Comment:
You should use `at::full(value, size, options)`, it's specifically to fill a tensor with single value